ci/test/00_setup_env_win64.sh

```diff
@@ -7,13 +7,10 @@
 export LC_ALL=C.UTF-8
 
 export CONTAINER_NAME=ci_win64
-export CI_IMAGE_NAME_TAG="docker.io/amd64/debian:bookworm"  # Check that https://packages.debian.org/bookworm/g++-mingw-w64-x86-64-posix (version 12.2, similar to guix) can cross-compile
+export CI_IMAGE_NAME_TAG="docker.io/amd64/ubuntu:23.10"  # Check that https://packages.ubuntu.com/mantic/g++-mingw-w64-x86-64-posix (version 12.3, similar to guix)  can cross-compile
```

Comment:
It's needed if we want mingw-w64 11.x (which fixes the return type issues). Bookworm only ships 10. It could be dropped if we want to retain the workarounds.
Mantic: https://packages.ubuntu.com/mantic/mingw-w64-x86-64-dev
Bookworm: https://packages.debian.org/bookworm/mingw-w64-x86-64-dev

Reply:
Will add it's also a better match to Guix, as there we are using GCC 12.3.0 & 11.x headers, and that is what would be used on Mantic, vs GCC 12.2.0 and 10.x on Debian.

Reply:
> will be EOL in 3 months, so it is not an option. 

Is there a particular reason it can't be used in the interim, so we have a better matching CI and less workarounds to maintain? I assume we are going to switch to Noble when it releases in April, so Mantic going EOL in July isn't an issue?